### PR TITLE
(travis) refactor travis CI

### PR DESCRIPTION
**Pull request by odanoburu**
- add ubuntu, mac, and windows builds

### CODE REVIEW

**Comment by odanoburu** [comment]
I must say that the windows build doesn't seem very reliable because of the chocolatey server. I've had 3-4 failures that were caused by this (among ~40 builds). one can simply rebuild or check the log to see if this was cause, or we can tell travis that the windows build may fail.

File: Make.hs [hunk 1]
```diff
@@ -444,7 +444,6 @@ execute command args = do
 
 -- | For parallel RGL module compilation
 -- Unfortunately, this has no effect unless compiled with -threaded
-parallel_ :: (Foldable t, Monad m) => t (m a) -> m ()
```
Comment (johnjcamilleri) [inline Make.hs]: Could you comment this instead of deleting the line?

Comment (odanoburu) [reply Make.hs]: there you go!

Comment (odanoburu) [reply Make.hs]: I rebased the commit and force-pushed, so the github interface looks kind of strange

**Pull request merged**